ghcide/src/Development/IDE/GHC/ExactPrint.hs

```diff
@@ -81,24 +107,37 @@ data GetAnnotatedParsedSource = GetAnnotatedParsedSource
 
 instance Hashable GetAnnotatedParsedSource
 instance NFData GetAnnotatedParsedSource
+#if MIN_VERSION_ghc(9,2,0)
```

Comment:
Some of this CPP seems like it should maybe be in the ExactPrint compat module?

Reply:
Moving code around damages/loses history, so I'd like to avoid that. 
I have added module header comments to disambiguate the purpose of each module